src/v2/Apps/Onboarding/OnboardingApp.tsx

```diff
@@ -0,0 +1,108 @@
+import { FC } from "react"
+import { Text, Button, Flex, Spacer, Box } from "@artsy/palette"
+import {
+  OnboardingProvider,
+  useOnboardingContext,
+} from "./useOnboardingContext"
+import { useRouter } from "v2/System/Router/useRouter"
+import { OnboardingWelcome } from "./Views/OnboardingWelcome"
+import { OnboardingQuestionOne } from "./Views/OnboardingQuestionOne"
+import { OnboardingQuestionTwo } from "./Views/OnboardingQuestionTwo"
+import { OnboardingDone } from "./Views/OnboardingDone"
+import {
+  VIEW_AUCTION_HIGHLIGHTS,
+  VIEW_CURATED_ARTWORKS,
+  VIEW_SEARCH_ARTISTS,
+  VIEW_SEARCH_ARTWORKS,
+  VIEW_SEARCH_GALLERIES,
+  VIEW_TRENDING_ARTISTS,
+  VIEW_TRENDING_LOTS,
+} from "./config"
+import { OnboardingProgress } from "./Components/OnboardingProgress"
+
+export const OnboardingApp: FC = () => {
+  const { router } = useRouter()
+
+  const handleDone = () => {
+    router.push("/")
+  }
+
+  return (
+    <OnboardingProvider onDone={handleDone}>
+      <OnboardingDebug />
+
+      <Spacer mt={2} />
+
+      <OnboardingSteps />
+    </OnboardingProvider>
+  )
+}
+
+const OnboardingSteps: FC = () => {
+  const { current, next } = useOnboardingContext()
+
+  switch (current) {
+    case "Welcome":
+      return <OnboardingWelcome />
+
+    case "WhatDoYouLoveMost":
+      return (
+        <>
+          <OnboardingProgress />
+          <Spacer mt={1} />
+          <OnboardingQuestionOne />
+        </>
+      )
+
+    case "WhereWouldYouLikeToDiveIn":
+      return (
+        <>
+          <OnboardingProgress />
+          <Spacer mt={1} />
+          <OnboardingQuestionTwo />
+        </>
+      )
+
+    case "Done":
+      return <OnboardingDone />
+
+    case VIEW_AUCTION_HIGHLIGHTS:
+    case VIEW_CURATED_ARTWORKS:
+    case VIEW_SEARCH_ARTISTS:
+    case VIEW_SEARCH_ARTWORKS:
+    case VIEW_SEARCH_GALLERIES:
+    case VIEW_TRENDING_ARTISTS:
+    case VIEW_TRENDING_LOTS:
```

Comment:
So from here: We should be able to just make components like `OnboardingAuctionHighlights`, `OnboardingSearchGalleries`, etc, and just slot them in.

Reply:
None of those components need to know anything about the external state other than to call the `next` function to proceed.

Reply:
yup yup :)

Reply:
Will the switch `case`s above this be refactored into this format? While I lack a bit of context on this, its a bit confusing to see this form
```
case "WhereWouldYouLikeToDiveIn"
```
right alongside this form
```
case VIEW_AUCTION_HIGHLIGHTS:
```
Whats the difference?

Reply:
I concur that we should just use the const form for all of them. Makes it a little less error prone.